src/bosh_azure_cpi/lib/cloud/azure/models/vm_cloud_props.rb

```diff
@@ -57,6 +43,33 @@ def initialize(vm_properties, global_azure_config)
         ephemeral_disk_hash['size'],
         ephemeral_disk_hash['type']
       )
+      @caching = vm_properties.fetch('caching', 'ReadWrite')
+
+      @availability_zone = vm_properties['availability_zone']
+      unless @availability_zone.nil?
+        cloud_error('Virtual Machines deployed to an Availability Zone must use managed disks') unless global_azure_config.use_managed_disks
+        cloud_error("'#{@availability_zone}' is not a valid zone. Available zones are: #{AVAILABILITY_ZONES}") unless AVAILABILITY_ZONES.include?(@availability_zone.to_s)
```

Comment:
can we get the AVAILABILITY_ZONES from rest api? so the cpi not need to be updated to support the new zones.

Reply:
no afaik.

Reply:
> no afaik.

seems we can get the zones from the resource sku list api.

Reply:
You are right. Will file an issue to track it. We can implement it in future release.